### PR TITLE
check for dependencies

### PR DESCRIPTION
**Pull request by willemdh**
Some code should be added for apt and zypper, but this works fro yum:

```
./FireMotD -T Red
2017-12-08 19:32:09,238: FireMotD: Error: Unable to detect jq. Please install.
```

```
./FireMotD -v -T Red
2017-12-08 19:31:59,208: FireMotD: Info: Verbose execution.
2017-12-08 19:31:59,531: FireMotD: Info: Detected openssh-clients
2017-12-08 19:31:59,820: FireMotD: Info: Detected bc
2017-12-08 19:32:00,121: FireMotD: Info: Detected sysstat
2017-12-08 19:32:00,533: FireMotD: Error: Unable to detect jq. Please install.
```

Please test if it does'nt result in any issues on apt system.

### CODE REVIEW

**Comment by ThomDietrich** [comment]
Did you consider the command command?

```
command -v foo >/dev/null 2>&1 || { echo "I require foo but it's not installed.  Aborting." >&2; exit 1; }
```

**Comment by willemdh** [comment]
I'll give it a try. It seems like a better solution which is on independent. Tx

**Comment by willemdh** [comment]
Hmm it doesn't work for moreutils, nor openssh-clients and sysstat. I'll leave the package check for now for them (for yum). It will only run when saving the json.

File: FireMotD [hunk 1]
```diff
@@ -159,6 +159,22 @@ VersionComparison () {
   return 3
 }
 
+function IsInstalled {
+  if [[ -x "/usr/bin/yum" ]] ; then
+    YumExec=$(which yum 2>/dev/null)
+    if yum list installed "$@" >/dev/null 2>&1; then
+     true
```
Comment (ThomDietrich) [inline FireMotD]: indention not correct

File: FireMotD [hunk 2]
```diff
@@ -159,6 +159,22 @@ VersionComparison () {
...
+    else
+      false
+    fi
+  elif [[ -x "/usr/bin/zypper" ]] ; then
+    WriteLog Verbose Info "No $@ check for zypper available."
+    true
+  elif [[ -x "/usr/bin/apt-get" ]] ; then
+    WriteLog Verbose Info "No $@ check for apt available."
```
Comment (ThomDietrich) [inline FireMotD]:
```
if dpkg-query -l "$@" >/dev/null 2>&1; then echo "true"; else echo "false"; fi
```

tested for moreutils, sysstat and openssh-client. Not sure why this is still needed though.

File: FireMotD (hunk 1 above)
Comment (willemdh) [reply FireMotD]: ha yes I know the indention is different, but i switched from 4 spaces to 2 lately.. :)

File: FireMotD (hunk 2 above)
Comment (willemdh) [reply FireMotD]: If I remember correctly moreutils is needed for sponge, sysstat is need for mpstat. openssh-client I'm not sure anymore.. I'll leave that one out for now. Tx for the dpkg query I'll try integrate it

**Comment by willemdh** [comment]
So moreutils is needed for sponge, sysstat is need for mpstat. openssh-client I'm not sure anymore.. I'll leave that one out for now. Tx for the dpkg query integrated it, please confirm if it works

**Comment by willemdh** [comment]
Ok, I converted it to a command -v only version which is acturally quicker. Left the IsInstalled function as it could be useful.

File: FireMotD (hunk 1 above)
Comment (ThomDietrich) [reply FireMotD]: I saw that... but here it is 1 space.
Imho you should switch the whole script soon and use an editor supporting auto code formatting to bring everything in order.

**Pull request merged**